### DISCUSSION TITLE
Is there any way hiding functions from global context?

### DISCUSSION THREAD

**Discussion by junpark-vcanus** [jeffreylanters/react-unity-webgl#179] [Question and Answer]
First, this library is so amazing. You saved me. Thanks so much.

I tried unity webgl on very default way(create functions in script tag in index.html by unity build) first and now I'm using this(@7.x).
In both case, anyone can access the functions that I created via console in browser(You can just find a function by name you made or, in the case with this lib, react-unity object).
 
I think it could be a huge issue on security with my app(It might be using in industry).

I tried to figure out how this(unity WebGL) works and hide functions but I came to the conclusion that this is the limit of unity WebGL now. 

Do you think it's possible?

**Comment by jeffreylanters** [comment]
Hi Jun! First of all thanks!

This is a very interesting topic I haven't thought about before. I see the potential security issue, I could try to come up with an event system which do not expose the available events in the window's ReactUnityWebGL object. But this will break all current implementations.

But even if we manage to find a possible solution for this. Nothing is stopping users from invoking Unity messages by hand with or without this module. Do you have any suggestions? I'm going to think about this. Thanks for opening this discussion, I like having these kinds of conversations with fellow developers.
